internal/services/containerapps/container_app_resource_test.go

```diff
@@ -548,6 +563,95 @@ resource "azurerm_container_app" "test" {
 `, r.template(data), data.RandomInteger)
 }
 
+func (r ContainerAppResource) withKeyVaultSecret(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {
+    key_vault {
+      purge_soft_delete_on_destroy    = true
+      recover_soft_deleted_key_vaults = true
+    }
```

Comment:
Is this a must to set these features here in order to create the key vault? If not, then shall we remove the `noProvider` things scattered around to make things a bit simpler to review/maintain?

Reply:
Apologies - forgot to mention this in the PR. I based this on [how it's done for function apps](https://github.com/hashicorp/terraform-provider-azurerm/blob/7596b072d0b8bf04585d0b77facd8296640d8a2a/internal/services/appservice/linux_function_app_resource_test.go#L3659C1-L3667C4). I suppose it's just keeping the deployment side of things a bit cleaner - without this, the key vault would be in a soft-deleted state for 90 days by default, although we can reduce that to 7 days in the key vault config. what do you reckon?